extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultHealthCheck.java

```diff
@@ -40,51 +38,62 @@ public HashicorpVaultHealthCheck(HashicorpVaultClient client, Monitor monitor) {
             case INITIALIZED_UNSEALED_AND_ACTIVE -> {
-                result = HealthCheckResult.success();
+                // do nothing
             }
             case UNSEALED_AND_STANDBY -> {
-                var standbyMsg = format(HEALTH_CHECK_ERROR_TEMPLATE, "Vault is in standby", response.getPayload());
+                var standbyMsg = HEALTH_CHECK_UNSUCCESSFUL_TEMPLATE.formatted("Vault is in standby", response.getPayload());
                 monitor.warning(standbyMsg);
-                result = HealthCheckResult.failed(standbyMsg);
+                errors.add(standbyMsg);
             }
             case DISASTER_RECOVERY_MODE_REPLICATION_SECONDARY_AND_ACTIVE -> {
-                var recoveryModeMsg = format(HEALTH_CHECK_ERROR_TEMPLATE, "Vault is in recovery mode", response.getPayload());
+                var recoveryModeMsg = HEALTH_CHECK_UNSUCCESSFUL_TEMPLATE.formatted("Vault is in recovery mode", response.getPayload());
                 monitor.warning(recoveryModeMsg);
-                result = HealthCheckResult.failed(recoveryModeMsg);
+                errors.add(recoveryModeMsg);
             }
             case PERFORMANCE_STANDBY -> {
-                var performanceStandbyMsg = format(HEALTH_CHECK_ERROR_TEMPLATE, "Vault is in performance standby", response.getPayload());
+                var performanceStandbyMsg = HEALTH_CHECK_UNSUCCESSFUL_TEMPLATE.formatted("Vault is in performance standby", response.getPayload());
                 monitor.warning(performanceStandbyMsg);
-                result = HealthCheckResult.failed(performanceStandbyMsg);
+                errors.add(performanceStandbyMsg);
             }
             case NOT_INITIALIZED -> {
-                var notInitializedMsg = format(HEALTH_CHECK_ERROR_TEMPLATE, "Vault is not initialized", response.getPayload());
+                var notInitializedMsg = HEALTH_CHECK_UNSUCCESSFUL_TEMPLATE.formatted("Vault is not initialized", response.getPayload());
                 monitor.warning(notInitializedMsg);
-                result = HealthCheckResult.failed(notInitializedMsg);
+                errors.add(notInitializedMsg);
             }
             case SEALED -> {
-                var sealedMsg = format(HEALTH_CHECK_ERROR_TEMPLATE, "Vault is sealed", response.getPayload());
+                var sealedMsg = HEALTH_CHECK_UNSUCCESSFUL_TEMPLATE.formatted("Vault is sealed", response.getPayload());
                 monitor.warning(sealedMsg);
-                result = HealthCheckResult.failed(sealedMsg);
+                errors.add(sealedMsg);
             }
             default -> {
-                var unspecifiedMsg = format(HEALTH_CHECK_ERROR_TEMPLATE, "Unspecified response from vault. Code: " + response.getCode(), response.getPayload());
+                var unspecifiedMsg = HEALTH_CHECK_UNSUCCESSFUL_TEMPLATE.formatted("Unspecified response from vault. Code: " + response.getCode(), response.getPayload());
                 monitor.warning(unspecifiedMsg);
```

Comment:
maybe all those monitor invocations could be handled in _one_ single line after the switch statement?